internal/provider/node_room_member.go

```diff
@@ -365,3 +566,26 @@ func checkNodeExists(searchingForNodeName string, nodes *client.RoomNodes, reach
 	}
 	return false, ""
 }
+
+func checkNodeMembershipRule(ruleID string, rules []nodeRoomMembershipRule) (bool, int) {
+	for idx, rule := range rules {
+		if ruleID == rule.ID.ValueString() {
+			return true, idx
+		}
+	}
+	return false, -1
+}
+
+func checkNodeMembershipClause(resp []client.NodeMembershipClause, stateClauses []nodeRoomMembershipClause) bool {
+	for _, s := range stateClauses {
+		for _, r := range resp {
+			if r.Label == s.Label.ValueString() &&
+				r.Operator == s.Operator.ValueString() &&
+				r.Value == s.Value.ValueString() &&
+				r.Negate == s.Negate.ValueBool() {
+				return true
```

Comment:
[doubt] Probably I'm missunderstanding the purpose of this function, but it will return `true` if any of the clauses in `resp` exists in `state`, is that the expected behaviour, or we want to check that both slices contain exactly the same elements?

Reply:
The slice returned by the cloud should be compared with the state ones, one element by another. If there is a difference, it means that the   change should be reverted.